### ISSUE TITLE
Error thrown when using custom mail layout and template.

### ISSUE THREAD

**Issue by cajunpepper** [octobercms/october#5724]
I'm implementing some custom email templating into our October-based app. If the `system_mail_layouts`.`content_text` column is NULL or empty, an error is thrown in \modules\system\classes\MailManager.php on line 266. 

I suggest modifying it from this:
```
if ($template->layout)
```
to this:
```
if ($template->layout && !empty($template->layout->content_text))
```

Or maybe check if it is empty, and if so, default to '{{ content|raw }}' so it doesn't break.

Here's the full error that I was getting:
```
TypeError: Twig\Environment::createTemplate(): Argument #1 ($template) must be of type string, null given, called in \modules\system\classes\MailManager.php on line 304 and defined in \vendor\twig\twig\src\Environment.php:386
Stack trace:
#0 \modules\system\classes\MailManager.php(304): Twig\Environment->createTemplate(NULL)
#1 \modules\system\classes\MailManager.php(269): System\Classes\MailManager->parseTwig(NULL, Array)
#2 \modules\system\classes\MailManager.php(164): System\Classes\MailManager->renderTextTemplate(Object(System\Models\MailTemplate), Array)
#3 \modules\system\classes\MailManager.php(118): System\Classes\MailManager->addContentToMailerInternal(Object(Illuminate\Mail\Message), Object(System\Models\MailTemplate), Array, false)
#4 \modules\system\classes\MailManager.php(68): System\Classes\MailManager->addContentToMailer(Object(Illuminate\Mail\Message), 'cajunpepper.sha...', Array, false)
#5 \modules\system\ServiceProvider.php(599): System\Classes\MailManager->addContentFromEvent(Object(Illuminate\Mail\Message), 'cajunpepper.sha...', NULL, NULL, Array)
#6 \vendor\laravel\framework\src\Illuminate\Events\Dispatcher.php(421): System\ServiceProvider->System\{closure}(Object(October\Rain\Mail\Mailer), Object(Illuminate\Mail\Message), 'cajunpepper.sha...', Array, NULL, NULL)
#7 \vendor\laravel\framework\src\Illuminate\Events\Dispatcher.php(249): Illuminate\Events\Dispatcher->Illuminate\Events\{closure}('mailer.beforeAd...', Array)
#8 \vendor\october\rain\src\Events\PriorityDispatcher.php(94): Illuminate\Events\Dispatcher->dispatch('mailer.beforeAd...', Array, true)
#9 \vendor\october\rain\src\Events\PriorityDispatcher.php(61): October\Rain\Events\PriorityDispatcher->fireEvent('mailer.beforeAd...', Array, true)
#10 \vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php(338): October\Rain\Events\PriorityDispatcher->fire('mailer.beforeAd...', Array, true)
#11 \vendor\october\rain\src\Mail\Mailer.php(421): Illuminate\Support\Facades\Facade::__callStatic('fire', Array)
#12 \vendor\october\rain\src\Mail\Mailer.php(80): October\Rain\Mail\Mailer->addContent(Object(Illuminate\Mail\Message), 'cajunpepper.sha...', NULL, NULL, Array)
#13 \vendor\laravel\framework\src\Illuminate\Mail\Mailable.php(212): October\Rain\Mail\Mailer->send('cajunpepper.sha...', Array, Object(Closure))
#14 \vendor\laravel\framework\src\Illuminate\Support\Traits\Localizable.php(19): Illuminate\Mail\Mailable->Illuminate\Mail\{closure}()
#15 \vendor\laravel\framework\src\Illuminate\Mail\Mailable.php(213): Illuminate\Mail\Mailable->withLocale(NULL, Object(Closure))
#16 \vendor\laravel\framework\src\Illuminate\Mail\SendQueuedMailable.php(74): Illuminate\Mail\Mailable->send(Object(October\Rain\Mail\MailManager))
#17 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): Illuminate\Mail\SendQueuedMailable->handle(Object(October\Rain\Mail\MailManager))
#18 \vendor\laravel\framework\src\Illuminate\Container\Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#19 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#20 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod(Object(October\Rain\Foundation\Application), Array, Object(Closure))
#21 \vendor\laravel\framework\src\Illuminate\Container\Container.php(661): Illuminate\Container\BoundMethod::call(Object(October\Rain\Foundation\Application), Array, Array, NULL)
#22 \vendor\laravel\framework\src\Illuminate\Bus\Dispatcher.php(128): Illuminate\Container\Container->call(Array)
#23 \vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(141): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(Illuminate\Mail\SendQueuedMailable))
#24 \vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Mail\SendQueuedMailable))
#25 \vendor\laravel\framework\src\Illuminate\Bus\Dispatcher.php(132): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#26 \vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(124): Illuminate\Bus\Dispatcher->dispatchNow(Object(Illuminate\Mail\SendQueuedMailable), false)
#27 \vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(141): Illuminate\Queue\CallQueuedHandler->Illuminate\Queue\{closure}(Object(Illuminate\Mail\SendQueuedMailable))
#28 \vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Mail\SendQueuedMailable))
#29 \vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(126): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#30 \vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(70): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\Queue\Jobs\DatabaseJob), Object(Illuminate\Mail\SendQueuedMailable))
#31 \vendor\laravel\framework\src\Illuminate\Queue\Jobs\Job.php(98): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\DatabaseJob), Array)
#32 \vendor\laravel\framework\src\Illuminate\Queue\Worker.php(425): Illuminate\Queue\Jobs\Job->fire()
#33 \vendor\laravel\framework\src\Illuminate\Queue\Worker.php(375): Illuminate\Queue\Worker->process('database', Object(Illuminate\Queue\Jobs\DatabaseJob), Object(Illuminate\Queue\WorkerOptions))
#34 \vendor\laravel\framework\src\Illuminate\Queue\Worker.php(173): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\DatabaseJob), 'database', Object(Illuminate\Queue\WorkerOptions))
#35 \vendor\laravel\framework\src\Illuminate\Queue\Console\WorkCommand.php(148): Illuminate\Queue\Worker->daemon('database', 'wms_mail,mail', Object(Illuminate\Queue\WorkerOptions))
#36 \vendor\laravel\framework\src\Illuminate\Queue\Console\WorkCommand.php(131): Illuminate\Queue\Console\WorkCommand->runWorker('database', 'wms_mail,mail')
#37 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): Illuminate\Queue\Console\WorkCommand->handle()
#38 \vendor\laravel\framework\src\Illuminate\Container\Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#39 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#40 \vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod(Object(October\Rain\Foundation\Application), Array, Object(Closure))
#41 \vendor\laravel\framework\src\Illuminate\Container\Container.php(661): Illuminate\Container\BoundMethod::call(Object(October\Rain\Foundation\Application), Array, Array, NULL)
#42 \vendor\laravel\framework\src\Illuminate\Console\Command.php(183): Illuminate\Container\Container->call(Array)
#43 \vendor\symfony\console\Command\Command.php(291): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#44 \vendor\laravel\framework\src\Illuminate\Console\Command.php(153): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#45 \vendor\symfony\console\Application.php(1014): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#46 \vendor\symfony\console\Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#47 \vendor\symfony\console\Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#48 \vendor\laravel\framework\src\Illuminate\Console\Application.php(102): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#49 \vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php(155): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#50 \artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#51 {main}
```

**Comment by daftspunk** [comment]
This has been fixed in the next patch  (v3.3.12). Went with:

```
if ($template->layout && $template->layout->content_text) {
```

Thanks!

**Comment by cajunpepper** [comment]
Rad! Thanks for addressing the issue so quickly.
